lib/fog/azurerm/requests/compute/attach_data_disk_to_vm.rb

```diff
@@ -7,26 +7,25 @@ def attach_data_disk_to_vm(disk_params, async)
           # Variable un-packing for easy access
           vm_name = disk_params[:vm_name]
           vm_resource_group = disk_params[:vm_resource_group]
-          disk_name = disk_params[:disk_name]
+          disk_names = disk_params[:disk_name]
           disk_resource_group = disk_params[:disk_resource_group]
           disk_size = disk_params[:disk_size_gb]
           storage_account_name = disk_params[:storage_account_name]
 
-          msg = "Attaching Data Disk #{disk_name} to Virtual Machine #{vm_name} in Resource Group #{vm_resource_group}"
+          disk_names = disk_names.is_a?(Array) ? disk_names : [disk_names]
+
+          msg = "Attaching Data Disk #{disk_names.join(', ')} to Virtual Machine #{vm_name} in Resource Group #{vm_resource_group}"
           Fog::Logger.debug msg
           vm = get_virtual_machine_instance(vm_resource_group, vm_name)
           lun = get_logical_unit_number(vm.storage_profile.data_disks)
```

Comment:
```suggestion
          lun = get_logical_unit_number(vm.storage_profile.data_disks)
```
This line is causing your PR to fail. Since in your code, you are getting the value of `LUN` only once, it throws an exception when it tries to add a new data disk to that same `LUN`.